supportedBackends/htcondor/src/main/scala/cromwell/backend/impl/htcondor/caching/provider/mongodb/MongoCacheActor.scala

```diff
@@ -0,0 +1,89 @@
+package cromwell.backend.impl.htcondor.caching.provider.mongodb
+
+import com.mongodb.DBObject
+import com.mongodb.casbah.MongoCollection
+import com.mongodb.casbah.commons.{MongoDBObject, TypeImports}
+import com.mongodb.util.JSON
+import cromwell.backend.BackendJobExecutionActor.SucceededResponse
+import cromwell.backend.impl.htcondor.caching.CacheActor
+import cromwell.backend.impl.htcondor.caching.exception.{CachedResultAlreadyExistException, CachedResultNotFoundException}
+import cromwell.backend.impl.htcondor.caching.model.CachedExecutionResult
+import cromwell.backend.impl.htcondor.caching.provider.mongodb.model.{KryoSerializedObject, MongoCachedExecutionResult}
+import cromwell.backend.impl.htcondor.caching.provider.mongodb.serialization.KryoSerDe
+
+class MongoCacheActor(collection: MongoCollection,
+                      override val cacheEnabled: Boolean = true,
+                      override val forceRewrite: Boolean = false) extends CacheActor with KryoSerDe {
+
+  import cromwell.backend.impl.htcondor.caching.provider.mongodb.model.MongoCachedExecutionResultProtocol._
+  import spray.json._
+
+  val ErrMsg = "Got an exception when storing execution result for hash {}."
+  val HashIdentifier = "hash"
+  override val tag = s"[MongoCacheActor]"
+
+  override def readExecutionResult(hash: String): CachedExecutionResult = {
+    val query = MongoDBObject(HashIdentifier -> hash)
+    val result = collection.findOne(query)
+    result match {
+      case Some(mongoDbObject) =>
+        if (forceRewrite) throwCachedResultNotFoundException(hash)
+        else deserializeSucceededResponse(mongoDbObject)
+      case None => throwCachedResultNotFoundException(hash)
+    }
+  }
+
+  override def storeExecutionResult(cachedExecutionResult: CachedExecutionResult): Unit = {
+    try {
+      readExecutionResult(cachedExecutionResult.hash)
+      lazy val warnMsg = s"$tag Execution result hash {${cachedExecutionResult.hash}} is already defined in database."
```

Comment:
This doesn't need to be lazy since it'll be evaluated anyway in the line following it.